chore: Add `kn` usage to Getting Started guide (#2531)
knative/docs: docs/serving/getting-started-knative-app.md

[Learn more about image tag resolution.](./tag-resolution.md)

The Hello World sample app reads in an `env` variable, `TARGET`, then prints "Hello World: \${TARGET}!". If `TARGET` isn't defined, it will print "NOT SPECIFIED".

## Creating your Deployment with the Knative CLI

> The easiest way to create Knative service deployment is by using the Knative CLI [kn](https://github.com/knative/client).

Reviewer: ```suggestion
The easiest way to deploy a Knative Service is by using the Knative CLI [kn](https://github.com/knative/client).
```

> You can install the `kn` binary as described in [Installing the Knative CLI](../install/install-kn.md)

Reviewer: ```suggestion

**Prerequisite:** install the `kn` binary as described in [Installing the Knative CLI](../install/install-kn.md)
```

> `kn` allows you to create a service interactively on the command line.

Reviewer: ```suggestion
`kn` provides a command-line mechanism to create or update a Service, and will fill in defaults or prompt for missing parameters.
```

Author: Agreeed, that it should be clarified a bit and also "interactively" is misleading, as there is no real interaction or user dialog (so `kn` won't ask any questions). Let's skip that.

Proposal:

-------

It will create a corresponding resource description internally as when using a YAML file directly.
`kn` provides a command-line mechanism for managing Services.
It allows you to configure every aspect of a Service.
The only mandatory flag for creating a Service is `--image` with the container image reference as value.

-------

Reviewer: Sure, SGTM.

Mostly, I wanted to highlight that "interactive" suggests more of a wizard-style interface than I think `kn` aims for.

> Eventually, it will create a corresponding resource description internally as when using a YAML file directly.

Reviewer: I don't understand what this sentence ("Eventually, ...") means. Can we remove it?

> To create our service directly at the cluster, use:

Reviewer: ```suggestion
To create a Service directly at the cluster, use:
```

```shell
# Create a Knative service with the Knatice CLI kn

> kn service create helloworld-go --image gcr.io/knative-samples/helloworld-go --env TARGET="Go Sample v1"

Reviewer: The other section mentions that you can reference a different image in the YAML. Do you think we need similar text here?

Author: Good point, I will copy that over to the `kn` section.

Author: BTW, the section for the YAML files is

```
If you want to deploy the sample app, leave the config file as-is. If you're
deploying an image of your app, update the name of the app and the URL of
the image accordingly.
```

If ok, I will fix that, too, as the image value is not really an URL but a reference to a container image in 'Docker syntax'

Author: Added now the following to the YAML section:

-----

If you want to deploy the sample app, leave the config file as-is. If you're
deploying an image of your app, update the name of the Service (`.metadata.name`) and the reference to the container image (`.spec.containers[].image`) accordingly.

-----

+ the image comment in the yaml

```

Now that you have deployed the service, Knative will perform the following steps:

- Create a new immutable revision for this version of the app.
- Perform network programming to create a route, ingress, service, and load
  balancer for your app.
- Automatically scale your pods up and down based on traffic, including to zero
  active pods.

## Creating your Deployment with YAML

Alternatively, to deploy an app using Knative, you can also create the configuration in a YAML file that defines a service. For more information about the Service object, see the
[Resource Types documentation](https://github.com/knative/serving/blob/master/docs/spec/overview.md#service).

> If a configuration file is used, it specifies metadata about the application, points to the hosted image of the app for deployment, and allows the deployment to be

Reviewer: I think it's fine to leave this paragraph as it was.

```

If you want to deploy the sample app, leave the config file as-is. If you're
deploying an image of your own app, update the name of the app and the URL of
deploying an image of your app, update the name of the app and the URL of
the image accordingly.

## Deploying your app

From the directory where the new `service.yaml` file was created, apply the
configuration:

```bash
kubectl apply --filename service.yaml
```

> Now that your service is created, Knative will perform the following steps:

Reviewer: I think we still want this text here, as these steps still happen for both YAML and `kn` usage.

If you want, you can use the tabbed content to unify the common elements:
```
{{< tabs name="create" default="kn">}}
{{% tab name="kn" %}}
...
{{< /tab >}}
{{% tab name="kubectl" %}}
...
{{< /tab >}}
{{< /tabs >}}
```

I think it's fine to lay one out over the other, though.

Author: You mean to use dedicated tabs for the "Creating your Deployment with the Knative CLI" and "Creating your Deployment with YAML" and then only one sectuon "Creating your Deployment" ? 

For this intro case I tend to lay them out one over the other (as is), as the second tab can be easily overlooked and I think its important to prominently present both alternative.

For other sample in the documentation (which I will continue to work on), the tabbed approach makes totally sense.

For now I just added back the list with the steps performed  to the YAML section, too.

1. To find the URL for your service, enter:

>    ```shell

Reviewer: I think tabs would make this part look a lot nicer.

Author: Let me try that. Not sure how it looks like when rendered as it is complex content.

Reviewer: I've kicked off a preview. It's apparently triggered by `/lgtm` or `approve` labels, so I'm putting a `/hold` in, too.
